src/functions/assertions/HaveParameter.ps1

```diff
@@ -31,77 +31,50 @@
     }
 
     function Get-ParameterInfo {
-        param(
-            [Parameter( Mandatory = $true )]
+        param (
+            [Parameter(Mandatory = $true)]
             [Management.Automation.CommandInfo]$Command
         )
         <#
         .SYNOPSIS
-            Use Tokenize to get information about the parameter block of a command
+            Use AST to get information about the parameter block of a command
         .DESCRIPTION
             In order to get information about the parameter block of a command,
             several tools can be used (Get-Command, AST, etc).
             In order to get the default value of a parameter, AST is the easiest
-            way to go; but AST was only introduced with PSv3.
-            This function creates an object with information about parameters
-            using the Tokenize
+            way to go
         .NOTES
-            Author: Chris Dent
+            Author: Brian West
         #>
 
-        function Get-TokenGroup {
-            param(
-                [Parameter( Mandatory = $true )]
-                [System.Management.Automation.PSToken[]]$tokens
-            )
-            $i = $j = 0
-            do {
-                $token = $tokens[$i]
-                if ($token.Type -eq 'GroupStart') {
-                    $j++
-                }
-                if ($token.Type -eq 'GroupEnd') {
-                    $j--
-                }
-                if (-not $token.PSObject.Properties.Item('Depth')) {
-                    $token.PSObject.Properties.Add([Pester.Factory]::CreateNoteProperty("Depth", $j))
-                }
-                $token
-
-                $i++
-            } until ($j -eq 0 -or $i -ge $tokens.Count)
-        }
-
-        $errors = $null
-        $tokens = [System.Management.Automation.PSParser]::Tokenize($Command.Definition, [Ref]$errors)
-
         # Find param block
-        $start = $tokens.IndexOf(($tokens | & $SafeCommands['Where-Object'] { $_.Content -eq 'param' } | & $SafeCommands['Select-Object'] -First 1)) + 1
-        $paramBlock = Get-TokenGroup $tokens[$start..($tokens.Count - 1)]
-
-        for ($i = 0; $i -lt $paramBlock.Count; $i++) {
-            $token = $paramBlock[$i]
+        $ast = $Command.ScriptBlock.Ast
+
+        if ($null -ne $ast) {
+            $paramBlock = $ast.FindAll(
```

Comment:
For improved readability, we can return early to avoid unnecessary nesting of the remaining code.

```ps1
        $ast = $Command.ScriptBlock.Ast
        # Unsupported command-type. Unable to determine default value, so returning early
        if ($null -eq $ast) { return }

        $paramBlock = $ast.FindAll(
```

Reply:
I decided to return if not able to find the parameters.  I hope that's okay.